service-app/internal/factory/document_component.go

```diff
@@ -43,6 +45,27 @@ var componentRegistry = map[string]Component{
 	"LPA120": {
 		Parser: lpa120_parser.Parse,
 	},
+	constants.DocumentTypeLP2: {
+		Parser: generic_parser.Parse,
+	},
+	constants.DocumentTypeLPA002: {
+		Parser: generic_parser.Parse,
+	},
+	constants.DocumentTypeLPAPA: {
+		Parser: generic_parser.Parse,
+	},
+	constants.DocumentTypeLPAPW: {
+		Parser: generic_parser.Parse,
+	},
+	constants.DocumentTypeLPA114: {
+		Parser: generic_parser.Parse,
+	},
+	constants.DocumentTypeLPA117: {
+		Parser: generic_parser.Parse,
+	},
+	constants.DocumentTypeEP2PG: {
+		Parser: generic_parser.Parse,
+	},
 }
```

Comment:
We could introduce a map initialiser to avoid duplicates where possible e.g. a new function 

```
SupportedDocTypes := []string{
    constants.DocumentTypeLP2,
   ...
}

for _, docType := range SupportedDocTypes {
    componentRegistry[docType] = struct {
        Parser func(string) error
    }{
        Parser: generic_parser.Parse,
    }
}
```